<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <group ns="earth_rover">
        <node name="earth_rover_lidar_guard" pkg="earth_rover_lidar_guard" type="earth_rover_lidar_guard_node" output="screen" required="true">
            <param name="laser_topic_name" value="/rplidar/scan" type="str"/>
            <param name="odom_topic_name" value="/earth_rover/odom" type="str"/>
            <param name="max_left_speed_mps" value="0.75" type="double"/>
            <param name="max_right_speed_mps" value="0.75" type="double"/>
            <param name="wheel_distance" value="0.391266" type="double"/>
            <!-- <param name="angle_tf_deg" value="180.0" type="double"/> -->

            <rosparam param="bounding_polygon">
                [
                    -0.16426449967548254, 0.263175162486732,
                    -0.16426449967548254, 0.1832176214084029,
                    -0.20851720636710527, 0.1832176214084029,
                    -0.2260907995514572, 0.16564402822405097,
                    -0.22609079955145714, -0.16564402822405103,
                    -0.20851720636710522, -0.18321762140840295,
                    -0.16426449967548248, -0.18321762140840295,
                    -0.16426449967548248, -0.263175162486732,
                    -0.14669090649113056, -0.2807487556710839,
                    0.09669090621173385, -0.2807487556710839,
                    0.11426449939608577, -0.263175162486732,
                    0.11426449939608577, -0.21678492659702894,
                    0.5130901872180402, -0.21678492659702892,
                    0.5306637804023921, -0.199211333412677,
                    0.5306637804023921, 0.1992113334126771,
                    0.5130901872180402, 0.21678492659702903,
                    0.11426449939608571, 0.216784926597029,
                    0.11426449939608571, 0.263175162486732,
                    0.09669090621173379, 0.2807487556710839,
                    -0.14669090649113062, 0.2807487556710839
                ]
            </rosparam>
        </node>
    </group>
</launch>
